pennylane/_qubit_device.py

```diff
@@ -335,6 +336,43 @@ def execute(self, circuit, **kwargs):
             self.tracker.record()
         return results
 
+    def execute_shadows(self, circuit, **kwargs):
+        self.check_validity(circuit.operations, circuit.observables)
+
+        shadow_observable = circuit.observables[0]
+
+        rotations = []
+        samples = []
+        for obs in [qml.PauliX, qml.PauliY, qml.PauliZ]:
```

Comment:
Potentially stupid question: When measuring in X, Y or Z basis, shouldnt you apply Hadamard, Hadamard@PhaseShift(pi/2), and idendity?

Reply:
This used to be done in `execute_shadows`. I just made a simplification to move all of this logic to this function.